elephant/spike_train_surrogates.py

```diff
@@ -347,15 +347,15 @@ def dither_spike_train(spiketrain, shift, n=1, decimals=None, edges=True):
     if edges is False:
         # Move all spikes outside [spiketrain.t_start, spiketrain.t_stop] to
         # the range's ends
-        surr = np.minimum(np.maximum(surr.base,
-            (spiketrain.t_start / spiketrain.units).base),
-            (spiketrain.t_stop / spiketrain.units).base) * spiketrain.units
+        surr = np.minimum(np.maximum(surr.simplified.magnitude,
+            spiketrain.t_start.simplified.magnitude),
+            spiketrain.t_stop.simplified.magnitude) * pq.s
     else:
         # Leave out all spikes outside [spiketrain.t_start, spiketrain.t_stop]
-        tstart, tstop = (spiketrain.t_start / spiketrain.units).base,\
-                        (spiketrain.t_stop / spiketrain.units).base
-        surr = [s[np.all([s >= tstart, s < tstop], axis=0)] * spiketrain.units
-                for s in surr.base]
+        tstart, tstop = spiketrain.t_start.simplified.magnitude, \
+                        spiketrain.t_stop.simplified.magnitude
+        surr = [np.sort(s[np.all([s >= tstart, s < tstop], axis=0)]) * pq.s
```

Comment:
I would not add np.sort() here, since the method is only supposed to shift the original spike train, no matter how nicely constructed it was in the first place. So it should never change the order of spikes - this could affect some analyses, couldn't it?

Reply:
Sorting is necessary because the it is not automatically checked by the neo.Spiketrains class, but other functions (e.g. isi()) assumes the spike are in increasing order.

Reply:
Actually, a `sort()` for ISIs was already included in the meantime.